app/controllers/admin/dashboard_controller.rb

```diff
@@ -0,0 +1,41 @@
+class Admin::DashboardController < ApplicationController
+  after_action -> { flash.discard }, if: -> { request.xhr? }
+  
+  def index
+    authorize current_user, policy_class: Admin::DashboardPolicy
+  end
+
+  def pending_shops 
+    @pending_shops = Shop.where(state: "pending").includes(:user)
+    authorize @pending_shops, policy_class: Admin::DashboardPolicy
+  end
+
+  def handle_shop
+    @shop = Shop.find(params[:shop_id])
```

Comment:
Can't make sure @shop is a pending one.

Should change to 
```
Shop.pendings.find(params[:shop_id])
```

and add scope `pendings` to Shop model for re-use

Reply:
noted and updated